samcli/commands/_utils/options.py

```diff
@@ -43,8 +44,10 @@ def get_or_default_template_file_name(ctx, param, provided_value, include_build)
             if os.path.exists(option):
                 provided_value = option
                 break
-
     result = os.path.abspath(provided_value)
+
+    if ctx:
+        setattr(ctx, "config_path", os.path.dirname(provided_value))
     LOG.debug("Using SAM Template at %s", result)
```

Comment:
Clever! But you need to use 'result' and not provided_value because there could be relative paths that's provided

Reply:
Good catch! done.